### PR TITLE
Rspec matchers

### PR DESCRIPTION
**Pull request by ydaniju**

### CODE REVIEW

File: rspec_matchers.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    --require spec_helper      # requires spec_helper
+
+    Add a third:
+
+    -fd                 # the test will run with full documentation
+
+
+#### Conclusion
+
+We can do more of this next time. At the moment, enjoy your tests.
```
Comment (codesword) [inline rspec_matchers.md:138]: :+1: Very good writeup. Short and on point. 

I noticed there is no section on why I need to create my own matcher. That could be in the conclusion though. What do you think?

Comment (ydaniju) [inline rspec_matchers.md:138]: Good point G.O. I'm on it.

Comment (ydaniju) [inline rspec_matchers.md:138]: Done @andela-iokonkwo.

File: rspec_matchers.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
+# Creating your own RSpec Matchers
+
+RSpec is a behavior-driven development (BDD) framework for the Ruby programming language. It assists with setting expectations for how exactly one expects ones code to behave. Tests (or specs) are first written and run leading to failures. Code is then written to pass those expectations earlier set.
+
+RSpec by default provides a handful of matchers that can help you test your code. Matchers are sets of expressions which are used to compare between actual and expected values. The actual value is what is returned from the written code while the expected value is the value that satisfies the condition in the matcher.
+
+### Why write your own matchers
+
+Sometimes you might hve written your code and you could not find a straightforward matcher to test it. An example is in Rails Application when you like to test if a model class ```User``` has many ```Microposts``` i.e
```
Comment (codesword) [inline rspec_matchers.md]: hve =>  have

**Comment by codesword** [comment]
This PR is good to go. However it was supposed to be merged with `ready` branch, not master branch. I will be closing it now. Please open a new PR against `ready` branch. 
Also don't forget to send your author bios to @unicodeveloper.

**Pull request closed without merging**